Source/Scene/GlobeSurfaceTileProvider.js

```diff
@@ -438,10 +438,18 @@ define([
      * @returns {Visibility} The visibility of the tile.
      */
     GlobeSurfaceTileProvider.prototype.computeTileVisibility = function(tile, frameState, occluders) {
-        var surfaceTile = tile.data;
+        if (frameState.fogEnabled) {
+            var distance = this.computeDistanceToTile(tile, frameState);
+            var scalar = distance * frameState.fogDensity;
```

Comment:
Let's move the fog equation to a `@private` helper function.  Keeping it in this file is probably fine for now.